handler.go

```diff
@@ -145,6 +145,16 @@ func (h *Handler) serveQuery(w http.ResponseWriter, r *http.Request, u *User) {
 		return
 	}
 
+	// If authentication is enabled and there are no users yet, make sure
+	// the first statement is creating a new cluster admin.
+	if h.AuthenticationEnabled && h.server.UsersLen() == 0 {
```

Comment:
I agree, but this seems like a "required" field.  Leaving it to chance in code later on could lead to a security vulnerability.  This one feels like you should have to explicitly tell the server it is either doing auth or not.  I look at this is as an exception to the rule, only because its security related.

If test coverage changes, or we remove that line of code setting it, we are exposed.  If it's part of the function signature, it can never be forgotten.